### ISSUE TITLE
Request: Better Native API error when attempting to upload file while dataset is locked

### ISSUE THREAD

**Issue by matthew-a-dunlap** [IQSS/dataverse#8700]
Hello!

I'm building a file uploader to dataverse as part of [CORE2](https://github.com/OdumInstitute/dataverse-corere), and ran into a confusion experience using the Native API. 

Dataverse locks the dataset during ingest if a file needs to be converted to tabular. It would be helpful for the API to report this as the reason for the upload failing. As it stands, the error is a very generic "Failed to add file to dataset."

Thanks for reading :)

**Comment by qqmyers** [comment]
FWIW: There are a number of recent changes/additions in the API that might help you avoid this (not to argue against the upload api also being more informative):

- You can get the list of locks on the dataset
- You can get the set of update times for a dataset that would also allow you to wait for things like indexing to complete
- With direct upload, you can upload multiple files and then  inform Dataverse at the end to add all the files at once, avoiding having the ingest, indexing, etc. triggered after each file and potentially interfering with your next upload.

**Comment by pdurbin** [comment]
Yes, a more informative error would be great. I'm always advising people to check for locks when they run into this problem. Most recently it was @atrisovic for https://github.com/IQSS/dataverse-uploader and the code looks something like this:

        for f in files:
            df = Datafile()
            df.set({
                "pid" : args.doi,
                "filename" : f,
                "directoryLabel": root[5:],
                "description" : \
                  "Uploaded with GitHub Action from {}.".format(
                    args.repo),
                })
            resp = api.upload_datafile(
                args.doi, join(root,f), df.json())
            check_dataset_lock(5)

You'll find similar logic in create_sample_data.py at https://github.com/IQSS/dataverse-sample-data

**Comment by pdurbin** [comment]
@matthew-a-dunlap any interest in making a pull request for this? 😄
